test/bioprinttest.c

```diff
@@ -244,11 +244,42 @@ static int test_fp(int i)
 static int test_big(void)
 {
     char buf[80];
+    volatile double d, z, inf, nan;
```

Comment:
What about making z a non-static global variable instead?

Reply:
could work. I can give it a try.

Reply:
I will name it "zero_value" so it is less likely to collide with other globals.

Reply:
It will also need `extern zero_value;` to appease the compiler warnings.

Reply:
it compiles on linux this way, I  dont see the reason why?

Reply:
Ouch clang is silly.

Reply:
Clang isn't silly.  It's pedantic.  It's the only compiler I've found that produces good code for:
```C
unsigned magic(unsigned n, unsigned d, int *o) {
    unsigned int r = n * d;

    *o = r / d != n;
    return r;
}
```

Reply:
Hmm, "-Wmissing-variable-declarations" is only in clang_devteams waring,
gcc would also implement this warning, so maybe that should be enabled
for gcc (unless we don't want this warning)

Reply:
> Clang isn't silly. It's pedantic. It's the only compiler I've found that produces good code for:

I checked the generated code, it optimizes magic(0,0,&r)
unter the assumption that 0/0 == 0.

Reply:
More specifically, it omits the division entirely.